### PR TITLE
Zero out default config

### PR DESCRIPTION
**Pull request by pathornteng**
Zero out default config value, the default config will come from genesis file.

### CODE REVIEW

File: builtin/types/chainconfig/chainconfig.proto [hunk 1]
```diff
@@ -97,11 +98,11 @@ message Action {
 }
 
 message AppStoreConfig {
-    uint64 num_evm_keys_to_prune = 1;
+    uint64 num_evm_keys_to_prune = 1 [(gogoproto.jsontag) = "num_evm_keys_to_prune"];
 }
 
 message EvmConfig {
-    uint64 gas_limit = 1;
+    uint64 gas_limit = 1 [(gogoproto.jsontag) = "gas_limit"];
```
Comment (enlight) [inline builtin/types/chainconfig/chainconfig.proto]: What's this for?

File: config/config.go [hunk 2]
```diff
@@ -24,10 +23,10 @@ var (
 func DefaultConfig() *cctypes.Config {
 	return &cctypes.Config{
 		AppStore: &cctypes.AppStoreConfig{
-			NumEvmKeysToPrune: 50,
+			NumEvmKeysToPrune: 0,
 		},
 		Evm: &cctypes.EvmConfig{
-			GasLimit: math.MaxUint64,
+			GasLimit: 0,
```
Comment (enlight) [inline config/config.go]: Seeing as this is the default value for the data type what's the point of `DefaultConfig()`?

File: builtin/types/chainconfig/chainconfig.proto (hunk 1 above)
Comment (pathornteng) [reply builtin/types/chainconfig/chainconfig.proto]: To remove `omitempty` tag as we are parsing this protobuf to genesis json. With `omitempty` tag, 0 value will not be written out.

File: config/config.go (hunk 2 above)
Comment (pathornteng) [reply config/config.go]: This is for iniitializing `Evm` and `AppStore` config or it will be nil.

Comment (enlight) [reply config/config.go]: You can probably get accomplish the same thing by using the getter functions on the generated protobuf, they appear to deal with `nil`.

Comment (pathornteng) [reply config/config.go]: you cannot just pass uninitialised config like
```
var cfg3 cctypes.Config
SetConfigSetting(&cfg3, "AppStore.NumEvmKeysToPrune", "1000") 
```
We need a function that initialise the config, allocating memory for `SetConfigSetting` to set value to the memory.
https://github.com/loomnetwork/go-loom/blob/cdb4849c08fc2bf805b2fdcc7a6d3ac65c9affe7/config/config.go#L38

**Pull request merged**